packages/protocol/contracts/governance/Election.sol

```diff
@@ -857,4 +905,56 @@ contract Election is
     }
     return res;
   }
+
+  // Struct to hold local variables for `forceDecrementVotes`.
+  // Needed to prevent solc error of "stack too deep" from too many local vars.
+  struct DecrementVotesInfo {
+    address[] groups;
+    uint256 remainingValue;
+  }
+
+  /**
+   * @notice Reduces the total amount of `account`'s voting gold by `value` by
+   *         iterating over all groups voted for by account.
+   * @param account Address to revoke votes from.
+   * @param value Amount of votes to revoke.
+   * @param lessers The groups receiving fewer votes than the i'th `group`, or 0 if
+   *                the i'th `group` has the fewest votes of any validator group.
+   * @param greaters The groups receivier more votes than the i'th `group`, or 0 if
+   *                the i'th `group` has the most votes of any validator group.
+   * @param indices The indices of the i'th group in the account's voting list.
+   */
+  function forceDecrementVotes(
+    address account,
+    uint256 value,
+    address[] calldata lessers,
+    address[] calldata greaters,
+    uint256[] calldata indices
+  ) external nonReentrant onlyRegisteredContract(LOCKED_GOLD_REGISTRY_ID) returns (uint256) {
+    require(value > 0 && value <= getTotalVotesByAccount(account));
```

Comment:
getTotalVotesByAccount is an expensive check (iterates over all the voted groups). What if instead we go on an decrementVotes, and as a last step we check that remainingValue is 0. If it's not then there's a problem and we fail.

With this strategy we optimize for the valid calls (it's cheaper); and deoptimze for the invalid call (will fail after doing more work); which i think is a good strategy.